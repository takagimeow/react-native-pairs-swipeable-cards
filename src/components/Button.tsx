import React from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';

export const Button = ({
  name,
  color = '#c4c4c4',
  onPress,
}: {
  name: 'reply' | 'heart' | 'handshake-o';
  color?: string;
  onPress?: () => void;
}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.iconContainer}>
        <FontAwesome name={name} size={28} color={color} />
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#ffffff',
    elevation: 5,
    borderRadius: 50,
    shadowColor: 'rgba(196, 196, 196, 0.25)',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 2.5,
    shadowRadius: 7.4,
    paddingHorizontal: 8,
    paddingVertical: 8,
  },
  iconContainer: {
    width: 48,
    height: 48,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
