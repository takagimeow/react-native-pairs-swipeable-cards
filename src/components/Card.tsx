import React from 'react';
import { Animated, View, StyleSheet, Dimensions, PanResponderInstance, Text } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';

export const CARD_WIDTH = Dimensions.get('screen').width * 0.915;
export const CARD_HEIGHT = Dimensions.get('screen').height * 0.695;
export const BORDER_RADIUS = 12;

export interface ICard {
  children: JSX.Element;
  panResponder: PanResponderInstance;
  pan: Animated.ValueXY;
  opacity: Animated.Value;
  overlayOpacity: Animated.AnimatedInterpolation;
  overlayColor: Animated.AnimatedInterpolation;
  cardState: string;
  rotate: string | Animated.Value | Animated.AnimatedInterpolation;
  scale: number | Animated.Value | Animated.AnimatedInterpolation;
}

export const Card = ({
  children,
  panResponder,
  pan,
  opacity,
  overlayOpacity,
  overlayColor,
  cardState,
  rotate,
  scale,
}: ICard) => {
  return (
    <Animated.View
      {...panResponder.panHandlers}
      style={[
        pan.getLayout(),
        styles.box,
        {
          opacity,
          transform: [
            {
              rotate,
            },
            {
              scale: scale,
            },
          ],
        },
        {
          position: 'absolute',
        },
      ]}
    >
      <Animated.View
        style={[
          {
            position: 'absolute',
            top: 0,
            left: 0,
            width: '100%',
            height: '100%',
            backgroundColor: overlayColor,
            zIndex: 1,
            opacity: overlayOpacity,
            paddingHorizontal: 8,
            paddingVertical: 8,
            borderRadius: BORDER_RADIUS - 1,
          },
        ]}
      >
        <View
          style={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: cardState === 'good' ? 'flex-start' : 'flex-end',
            alignItems: 'center',
          }}
        >
          <View
            style={{
              marginRight: 8,
            }}
          >
            <FontAwesome
              name={cardState === 'good' ? 'thumbs-up' : 'reply'}
              size={40}
              color={'#ffffff'}
            />
          </View>
          <Text style={styles.text}>{cardState === 'good' ? 'いいね' : 'スキップ'}</Text>
        </View>
      </Animated.View>
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  box: {
    backgroundColor: '#f0f0f0',
    width: CARD_WIDTH,
    height: CARD_HEIGHT,
    borderRadius: BORDER_RADIUS,
    borderWidth: 1,
  },
  text: {
    fontSize: 48,
    fontWeight: 'bold',
    color: '#ffffff',
  },
});
