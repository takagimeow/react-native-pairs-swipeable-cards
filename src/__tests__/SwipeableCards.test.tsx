/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import { render } from '@testing-library/react-native';

import { SwipeableCards } from '../components/SwipeableCards';
import { BORDER_RADIUS } from '../components/Card';
import { ImageBackground, Text, View } from 'react-native';

jest.mock('@expo/vector-icons', () => {
  return {
    FontAwesome: () => (<View></View>) as any,
  };
});
const data: {
  id: string;
  isActive: boolean;
  name: string;
  age: number;
  place: string;
  rate: number;
  tags: string[];
  imageUri: string;
}[] = [
  {
    id: 'misono',
    isActive: true,
    name: '高村美園',
    age: 25,
    place: '神奈川県',
    rate: 64,
    tags: ['読書好き', '友達みたいな恋人が・・・', '気ままにドライブに行きたい!'],
    imageUri: 'https://pakutaso.cdn.rabify.me/shared/img/thumb/AMEMAN17826009.jpg.webp?d=1420',
  },
  {
    id: 'arisa',
    isActive: false,
    name: '須藤亜里沙',
    age: 24,
    place: '福岡県',
    rate: 64,
    tags: ['読書好き', '友達みたいな恋人が・・・', '気ままにドライブに行きたい!'],
    imageUri: 'https://pakutaso.cdn.rabify.me/shared/img/thumb/TS0529003.jpg.webp?d=1420',
  },
  {
    id: 'sasaki',
    isActive: false,
    name: '佐々木涼平',
    age: 24,
    place: '東京都',
    rate: 64,
    tags: ['読書好き', '友達みたいな恋人が・・・', '気ままにドライブに行きたい!'],
    imageUri:
      'https://pakutaso.cdn.rabify.me/shared/img/thumb/SSK_hinatawomitumeruretoribar.jpg.webp?d=710',
  },
  {
    id: 'tsugumi',
    isActive: true,
    name: '川本つぐみ',
    age: 24,
    place: '東京都',
    rate: 64,
    tags: ['読書好き', '友達みたいな恋人が・・・', '気ままにドライブに行きたい!'],
    imageUri: 'https://pakutaso.cdn.rabify.me/shared/img/thumb/nekocatIMGL54791978.jpg.webp?d=710',
  },
];

describe('SwipeableCards', () => {
  it('Snapshot Test', () => {
    const { toJSON } = render(
      <SwipeableCards
        data={data}
        onLike={() => console.log('いいね')}
        onSkip={() => console.log('スキップ!')}
        renderItem={(item, index) => (
          <ImageBackground
            key={`render_item_key_${index}`}
            source={{
              uri: item.imageUri,
            }}
            resizeMode="cover"
            style={{
              flex: 1,
              justifyContent: 'flex-end',
              paddingVertical: 16,
            }}
            imageStyle={{
              borderRadius: BORDER_RADIUS,
            }}
          >
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                paddingHorizontal: 16,
                marginBottom: 8,
              }}
            >
              <View
                style={{
                  width: 16,
                  height: 16,
                  borderRadius: 25,
                  backgroundColor: item.isActive ? '#2cd26b' : '#cc0033',
                  marginRight: 8,
                }}
              />
              <Text
                style={{
                  color: '#ffffff',
                  fontWeight: 'bold',
                  fontSize: 32,
                }}
              >
                {item.name}
              </Text>
            </View>
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                paddingHorizontal: 16,
                marginBottom: 8,
              }}
            >
              <Text
                style={{
                  color: '#ffffff',
                  fontWeight: 'bold',
                  fontSize: 16,
                  marginRight: 10,
                }}
              >
                {`${item.age}歳`}
              </Text>
              <Text
                style={{
                  color: '#ffffff',
                  fontWeight: 'bold',
                  fontSize: 16,
                  marginRight: 10,
                }}
              >
                {`${item.place}`}
              </Text>
              <Text
                style={{
                  color: '#ffffff',
                  fontWeight: 'bold',
                  fontSize: 16,
                }}
              >
                {`♡${item.rate}%`}
              </Text>
            </View>
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                paddingHorizontal: 16,
              }}
            >
              {item.tags.map(tag => (
                <View
                  key={`tag_${tag}`}
                  style={{
                    paddingVertical: 8,
                    paddingHorizontal: 10,
                    borderRadius: 6,
                    marginRight: 6,
                    backgroundColor: '#ffffff',
                  }}
                >
                  <Text
                    style={{
                      color: '#000000',
                      fontSize: 8,
                      fontWeight: 'bold',
                    }}
                  >
                    {tag}
                  </Text>
                </View>
              ))}
            </View>
          </ImageBackground>
        )}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
