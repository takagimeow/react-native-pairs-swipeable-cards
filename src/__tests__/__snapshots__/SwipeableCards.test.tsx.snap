// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableCards Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "height": "auto",
        "marginBottom": 48,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "height": 927.1299999999999,
          "position": "relative",
          "width": 686.25,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 927.1299999999999,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
              Object {
                "scale": 0.925,
              },
            ],
            "width": 686.25,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderRadius": 11,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View />
            </View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 48,
                  "fontWeight": "bold",
                }
              }
            >
              スキップ
            </Text>
          </View>
        </View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingVertical": 16,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://pakutaso.cdn.rabify.me/shared/img/thumb/nekocatIMGL54791978.jpg.webp?d=710",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 12,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#2cd26b",
                  "borderRadius": 25,
                  "height": 16,
                  "marginRight": 8,
                  "width": 16,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              川本つぐみ
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              24歳
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              東京都
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              ♡64%
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                読書好き
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                友達みたいな恋人が・・・
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                気ままにドライブに行きたい!
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 927.1299999999999,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
              Object {
                "scale": 0.925,
              },
            ],
            "width": 686.25,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderRadius": 11,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View />
            </View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 48,
                  "fontWeight": "bold",
                }
              }
            >
              スキップ
            </Text>
          </View>
        </View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingVertical": 16,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://pakutaso.cdn.rabify.me/shared/img/thumb/SSK_hinatawomitumeruretoribar.jpg.webp?d=710",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 12,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#cc0033",
                  "borderRadius": 25,
                  "height": 16,
                  "marginRight": 8,
                  "width": 16,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              佐々木涼平
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              24歳
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              東京都
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              ♡64%
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                読書好き
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                友達みたいな恋人が・・・
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                気ままにドライブに行きたい!
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 927.1299999999999,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
              Object {
                "scale": 0.925,
              },
            ],
            "width": 686.25,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderRadius": 11,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View />
            </View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 48,
                  "fontWeight": "bold",
                }
              }
            >
              スキップ
            </Text>
          </View>
        </View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingVertical": 16,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://pakutaso.cdn.rabify.me/shared/img/thumb/TS0529003.jpg.webp?d=1420",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 12,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#cc0033",
                  "borderRadius": 25,
                  "height": 16,
                  "marginRight": 8,
                  "width": 16,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              須藤亜里沙
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              24歳
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              福岡県
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              ♡64%
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                読書好き
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                友達みたいな恋人が・・・
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                気ままにドライブに行きたい!
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 927.1299999999999,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
              Object {
                "scale": 1,
              },
            ],
            "width": 686.25,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderRadius": 11,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View />
            </View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 48,
                  "fontWeight": "bold",
                }
              }
            >
              スキップ
            </Text>
          </View>
        </View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingVertical": 16,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://pakutaso.cdn.rabify.me/shared/img/thumb/AMEMAN17826009.jpg.webp?d=1420",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 12,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#2cd26b",
                  "borderRadius": 25,
                  "height": 16,
                  "marginRight": 8,
                  "width": 16,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              高村美園
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 8,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              25歳
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              神奈川県
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              ♡64%
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                読書好き
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                友達みたいな恋人が・・・
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 6,
                  "marginRight": 6,
                  "paddingHorizontal": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 8,
                    "fontWeight": "bold",
                  }
                }
              >
                気ままにドライブに行きたい!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 32,
        "paddingHorizontal": 64,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 50,
          "elevation": 5,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
          "shadowColor": "rgba(196, 196, 196, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 2.5,
          "shadowRadius": 7.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          }
        }
      >
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 50,
          "elevation": 5,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
          "shadowColor": "rgba(196, 196, 196, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 2.5,
          "shadowRadius": 7.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          }
        }
      >
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 50,
          "elevation": 5,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
          "shadowColor": "rgba(196, 196, 196, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 2.5,
          "shadowRadius": 7.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
</View>
`;
